### ISSUE TITLE
:key? in topic config

### ISSUE THREAD

**Issue by DaveWM** [FundingCircle/jackdaw#5]
At the moment, the avro serde requires a key of `:key?` to be present in the topic config (see the code [here](https://github.com/FundingCircle/jackdaw/blob/d121850d151ef77e1ced33a8fc754be307667448/src/jackdaw/serdes/avro.clj#L593)). If the topic config doesn't include it, you get a NPE. AFAICT, this key determines whether the serde is for a key or value - so the topic config doesn't seem like the right place for it, as there can be an avro serde for both the key and the value. 

Am I missing something here? Or would it be best to pass `key?` as a separate parameter to `avro-serde`?

**Comment by arrdem** [comment]
You're absolutely correct. The Avro serde has seen a bunch of change lately to allow more general use, and it looks like the key? parameter fell through the cracks. Sorry about that. I'll try to get a patch together today.

**Comment by DaveWM** [comment]
no worries, thanks very much 👍
